examples/miscellaneous/plot_outlier_detection_bench.py

```diff
@@ -0,0 +1,201 @@
+        X = X[idx]  # reduce the sample size
+        y = y[idx]
+        lb = LabelBinarizer()
+        x1 = lb.fit_transform(X[:, 1].astype(str))
+        x2 = lb.fit_transform(X[:, 2].astype(str))
+        x3 = lb.fit_transform(X[:, 3].astype(str))
+        X = np.c_[X[:, :1], x1, x2, x3, X[:, 4:]]
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name == "http" or dataset_name == "smtp":
+        y = (y != b"normal.").astype(int)
+
+    if dataset_name == "forestcover":
+        dataset = fetch_covtype()
+        X = dataset.data
+        y = dataset.target
+        idx = rng.choice(X.shape[0], int(X.shape[0] * 0.1), replace=False)
+        X = X[idx]  # reduce the sample size
+        y = y[idx]
+
+        # inliers are those with attribute 2
+        # outliers are those with attribute 4
+        s = (y == 2) + (y == 4)
+        X = X[s, :]
+        y = y[s]
+        y = (y != 2).astype(int)
+
+    if dataset_name in ["glass", "wdbc", "cardiotocography"]:
+        dataset = fetch_openml(name=dataset_name, version=1, as_frame=False)
+        X = dataset.data
+        y = dataset.target
+
+        if dataset_name == "glass":
+            s = y == "tableware"
+            y = s.astype(int)
+
+        if dataset_name == "wdbc":
+            s = y == "2"
+            y = s.astype(int)
+            X_mal, y_mal = X[s], y[s]
+            X_ben, y_ben = X[~s], y[~s]
+
+            # downsampled to 39 points (9.8% outliers)
+            idx = rng.choice(y_mal.shape[0], 39, replace=False)
+            X_mal2 = X_mal[idx]
+            y_mal2 = y_mal[idx]
+            X = np.concatenate((X_ben, X_mal2), axis=0)
+            y = np.concatenate((y_ben, y_mal2), axis=0)
+
+        if dataset_name == "cardiotocography":
+            s = y == "3"
+            y = s.astype(int)
+
+    # 0 represents inliers, and 1 represents outliers
+    y = pd.Series(y, dtype="category")
+    Xs.append(X)
+    ys.append(y)
+
+# %%
+# Define outlier detection models
+# -------------------------------
+# There is no particular reason to choose algorithms LOF and IForest. The goal
+# is to show that different algorithm performs well on different datasets.
+
+from sklearn.neighbors import LocalOutlierFactor
+from sklearn.ensemble import IsolationForest
+
+models = [
+    ("LOF", LocalOutlierFactor(n_neighbors=20, contamination="auto")),
+    ("IForest", IsolationForest(random_state=rng, contamination="auto")),
+]
+
+# %%
+# Compute ROC curves
+# ------------------
+ys_pred = []
+pos_label = 0  # mean 0 belongs to positive class
+
+print("Estimators processing...")
+for i, dataset_name in enumerate(datasets_name):
+
+    ys_pred_ = []
+    for model_name, clf in models:
+        clf.fit(Xs[i])
+        if model_name == "LOF":
+            y_pred = clf.negative_outlier_factor_
+
+        if model_name == "IForest":
+            y_pred = clf.fit(Xs[i]).decision_function(Xs[i])
+
+        ys_pred_.append(y_pred)
+    ys_pred.append(ys_pred_)
```

Comment:
maybe make ys_pred a dictionary ``ys_pred["http"]["LOF"]`` ? it would be a bit more explicit

Reply:
Since I created a compute_prediction function. I will no longer use `ys_pred`